webapps-schema/src/main/java/io/camunda/webapps/schema/descriptors/AbstractIndexDescriptor.java

```diff
@@ -43,26 +51,30 @@ public String getMappingsClasspathFilename() {
 
   @Override
   public String getAllVersionsIndexNameRegexPattern() {
-    return String.format("%s-%s-\\d.*", getPrefixAndComponentName(), getIndexName());
+    if (hasGlobalPrefix) {
+      return String.format(
+          "%s%s-%s-\\d.*", formattedIndexPrefix(), getComponentName(), getIndexName());
+    } else {
+      return String.format("%s%s-\\d.*", formattedIndexPrefix(), getIndexName());
+    }
   }
 
   @Override
   public String getVersion() {
     return "1.0.0";
   }
 
+  private String formattedIndexPrefix() {
+    return getIndexPrefix() != null && !getIndexPrefix().isBlank() ? getIndexPrefix() + "-" : "";
+  }
+
   public String getIndexPrefix() {
     return indexPrefix;
   }
 
-  private String getPrefixAndComponentName() {
-    // Cannot start index with "-" so must not append "-" for empty prefix
-    final var prefix = getIndexPrefix().isBlank() ? getIndexPrefix() : getIndexPrefix() + "-";
-
-    // Legacy descriptors have the same index prefix as component name this avoids duplication.
-    return getIndexPrefix().contains(getComponentName())
-        ? getIndexPrefix()
-        : prefix + getComponentName();
+  public AbstractIndexDescriptor withHasGlobalPrefix(final boolean isGlobalPrefix) {
+    hasGlobalPrefix = isGlobalPrefix;
+    return this;
```

Comment:
❓ Couldn't this be derived from `getIndexPrefix() != null && !getIndexPrefix().isBlank()`

Reply:
I think we decided to always have the global prefix + component name even if they repeat stuff, I think this will cause issues in usages in the `operate` module as the FQN will change. 

The answer is it cannot. if it's not clear we can talk more.